### PR TITLE
Fix Mac Catalyst support of SPM

### PR DESCRIPTION
**Pull request by dahchon**
### Platforms Impacted
- [x] iOS
- [ ] visionOS
- [ ] macOS

### Description of changes

Add .macCatalyst for iOS source

### Binary change

N/A

### Verification

Create a new project of iOS, enable catalyst, then add this package

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/fluentui-apple/pull/1989)

### CODE REVIEW

**Comment by joannaquu** [comment]
#1878 has some discussion on why we've held off on catalyst support @anandrajeswaran for updates.

File: Package.swift
Comment (mischreiber) [inline Package.swift:26]: Could you fix the whitespace here? It should be the same as it was before the addition of `.macCatalyst`

**Review by mischreiber — COMMENTED** [review]
After talking offline to @anandrajeswaran, I believe we're at a place now where we'd be willing to take this change to unblock downstream clients. That said, we still do not intend to elevate Catalyst to a fully supported "platform" at this time, so our support will be highly limited.

**Review by mischreiber — APPROVED** [review]

**Comment by dahchon** [comment]
I'll need the approval of merge from main. thanks!

**Comment by mischreiber** [comment]
Thanks for the contribution!

**Pull request merged**
